File: #2419.longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {

        int n = nums.size();
        int max = *max_element(nums.begin(), nums.end());
        int result = 0;
        for (int i = 0; i < n; i++) {
            int cnt = 0;
            while (i < n && nums[i] == max) {
                cnt++;
                i++;
            }
            result = std::max(result, cnt);
        }
        return result;
    }
};

File: #2419.longest-subarray-with-maximum-bitwise-and.ts
function longestSubarray(nums: number[]): number {
    let max = 0;
    let count = 0;
    let maxCount = 0;
    for (const num of nums) {
        if (num > max) {
            max = num;
            count = 0;
            maxCount = 0;
        }
        count = num === max ? count + 1 : 0;
        maxCount = Math.max(maxCount, count);
    }
    return maxCount;
};
